### PR TITLE
Catchup and cleanup

### PR DESCRIPTION
**Pull request by gsbernstein**
- Copied updates from VidaUIKit (including aspect ratio helper)
 - Updated Swift
 - reorganized into several files
 - set identifier based on view class name (or take custom name)

### CODE REVIEW

File: constrain/Classes/AspectRatio.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+//  Created by Greg on 8/10/19.
+//
+
+import UIKit
+
+public extension Constraints {
+    
+    /// Apply an aspect ratio constraint to a view
+    /// Ratio is expressed as width/height
+    /// Checks that ratio is finite before applying (ag divide by zero errors)
```
Comment (gsbernstein) [inline constrain/Classes/AspectRatio.swift]:
```suggestion
    /// Checks that ratio is finite before applying (avoid divide by zero errors)
```

File: constrain/Classes/BaseTypes.swift [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        guard let view = view else {
+            print("View fell out of memory.")
+            return self
+        }
+        return applyAnchorConstraint(anchor1: view.widthAnchor, anchor2: anchor, identifier: .width, constant: constant, relationship: relationship, priority: priority)
+    }
+    
+}
+
+// C-c-c-c-combos
```
Comment (gsbernstein) [inline constrain/Classes/BaseTypes.swift]:
```suggestion
// Combos! Set more than one constraint at a time quickly.
```

File: constrain/Classes/Constraints.swift [hunk 3]
```diff
@@ -5,352 +5,132 @@
...
 /// The superview is used when no view is passed in and the superview's respective anchor is used when the anchor is omitted.
 public class Constraints {
 
-    private weak var view: UIView?
+    internal weak var view: UIView?
+    internal var viewName: String
     private var constraints: [ConstraintIdentifier: NSLayoutConstraint] = [:]
+    
+    public var latestConstraint: NSLayoutConstraint?
+    public var latestConstraints: ConstraintSet?
```
Comment (gsbernstein) [inline constrain/Classes/Constraints.swift:39]:
```suggestion
	// Not implemented yet
	@available(*, unavailable)
    public var latestConstraints: ConstraintSet?
```

**Review by anconaesselmann — APPROVED** [review]
Great, thanks!

**Pull request merged**